### DISCUSSION TITLE
Sniff exclude-pattern not respected anymore ?

### DISCUSSION THREAD

**Discussion by tomasnorre** [PHPCSStandards/PHP_CodeSniffer#680] [Q&A] [answered]
After adjusting (1) the Sniff according to https://github.com/PHPCSStandards/PHP_CodeSniffer/wiki/Coding-Standard-Tutorial#creating-the-sniff my `exclude-pattern` isn't respected anymore. 

```xml
<rule ref="src/Sniffs/StrictTypes/ExplainStrictTypesSniff.php">
  <exclude-pattern>*/*Test\.php</exclude-pattern>
  <exclude-pattern>*/.meta/*\.php</exclude-pattern>
  <exclude-pattern>src/*</exclude-pattern>
  <exclude-pattern>contribution/*.php</exclude-pattern>
</rule>
```

Any suggestions what i'm doing wrong here? It was working before without the `StrictTypes` directory, and with PHPCS 3.9.2

1) https://github.com/exercism/php/pull/857

**Comment by jrfnl** [comment]
@tomasnorre I suspect this may be a bug. I've just been running some tests and yes, it works without the `StrictTypes` directory, but I'm not sure why ;-)

Based on my current tests though, _with_ the `StrictTypes` directory, the excludes also didn't work on PHPCS 3.9.2. Looks like this was broken in PHPCS 3.5.0.

I'm taking a deeper look now.

**Comment by jrfnl** [comment]
Actually... the problem _used to be_ much bigger and 2/3 of the problem has already been solved in previous PHPCS versions.

This is a bug though, so could you please open a bug report for this ?

**Comment by jrfnl** [comment, marked as answer]
Follow up: https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/683#issuecomment-2478454674

Short answer (and solution): the included sniff _still_ does not comply with the PHPCS naming conventions. Once it does, things work fine.

**Comment by jrfnl** [comment]
@tomasnorre If I remember correctly, your issue is now solved by the sniff following the PHPCS naming conventions. Correct ? If so, can I close this discussion ?
